compiler/x/codegen/BinaryEvaluator.cpp

```diff
@@ -2001,54 +2063,54 @@ TR::Register *OMR::X86::TreeEvaluator::integerMulEvaluator(TR::Node *node, TR::C
          // are evaluated by the generic analyser call below
          if (targetRegister == 0 && IS_32BIT_SIGNED(value)) // decomposition failed
             {
-            TR_X86OpCodes opCode;
-            if (firstChild->getReferenceCount() > 1 ||
-                firstChild->getRegister() != 0)
+            if(isByteOperation)
```

Comment:
this seems like it would be dead now? to get here you need isLoadConst to be true, opCode != BADIA32Op and isByteOperation is true. I don't see how that can happen...

Reply:
if secondChild is an Int8 or byte then isByteOperation is true and OpCode is "opCode = isMemRef ? IMUL1AccMem : IMUL1AccReg;" ..... So those 2 conditions are true and then yes this code will be executed only for constants so isLoadConst should be true...

Reply:
In all cases except the case where value is more than 4 bytes, opCode will not be BADIA32Op. So opCode != BADIA32Op is true in all those cases.

Reply:
I think the `isByteOperation` branch is not needed here. The instructions got generated in this case are `IMUL1AccReg` and `IMUL1AccMem` which are the same instructions generated by the `temp.genericAnalyser` so we should just use the general analyser rather than a special case here.